File: LeetCode-2023/2379.得到-k-个黑块的最少涂色次数.py
#
# @lc app=leetcode.cn id=2379 lang=python3
#
# [2379] 得到 K 个黑块的最少涂色次数
#

# @lc code=start
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        cnt = ans = blocks.count('W', 0, k)
        for i in range(k, len(blocks)):
            cnt += blocks[i] == 'W'
            cnt -= blocks[i-k] == 'W'
            ans = min(ans, cnt)
        return ans
# @lc code=end

File: LeetCode-2023/2379.得到-k-个黑块的最少涂色次数.java
/*
 * @lc app=leetcode.cn id=2379 lang=java
 *
 * [2379] 得到 K 个黑块的最少涂色次数
 */

// @lc code=start
class Solution {
    public int minimumRecolors(String blocks, int k) {
        int cnt = 0;
        for(int i = 0; i < k; i++) {
            if(blocks.charAt(i) == 'W') {
                cnt++;
            }
        }
        int ans = cnt;
        for(int i = k; i < blocks.length(); i++) {
            cnt += blocks.charAt(i) == 'W' ? 1 : 0;
            cnt -= blocks.charAt(i-k) == 'W' ? 1 : 0;
            ans = Math.min(ans, cnt);
        }
        return ans;
    }
}
// @lc code=end

